src/aliceVision/sfmData/colorize.cpp

```diff
@@ -12,17 +12,51 @@
 #include <aliceVision/stl/mapUtils.hpp>
 #include <aliceVision/image/io.hpp>
 #include <aliceVision/system/ProgressDisplay.hpp>
+#include <aliceVision/system/Timer.hpp>
 
 #include <map>
 #include <random>
 #include <vector>
 #include <functional>
 namespace aliceVision {
+
+class LMColorAccumulator
+{
+    public:
+        LMColorAccumulator()
+            : rgbFinal(0.0)
+            , sumSegments(0.0)
+            {}
+        ~LMColorAccumulator() {}
+
+
+        image::RGBfColor rgbFinal;
+        double sumSegments;
+
+        void addRGB(const image::RGBfColor& rgbf, const double& weight) 
```

Comment:
for primitive types just pass by value
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const
```suggestion
        void addRGB(const image::RGBfColor& rgbf, double weight) 
```